packages/graphqlsp/src/ast/index.ts

```diff
@@ -90,6 +90,13 @@ function unrollFragment(
   // If we found another identifier, we repeat trying to find the original
   // fragment definition
   if (ts.isIdentifier(found)) {
+    // NOTE: A gotcha of `getDefinitionAtPosition` is that when resolution fails, it points
+    // back at the original identifier (e.g. if its of the internal type `error`).
+    if (found === element) {
+      // TODO: Instead of bailing out here, we should be able to issue a diagnostic here
+      // that tells the user that resolution of a fragment has failed.
+      return fragments;
```

Comment:
Let's add a `throws` argument or something so we can throw if need be

Reply:
re. notes in PR description, I don't think this is really safe to merge, but instead more for "notes" so I can come back to it 😄
There's several cases where I think similar issues can happen, and there's not really a great option to addressing most of them right now, until we can escalate granular errors, otherwise we basically end up either hiding the issue from the user and it'll only become apparent when they check output (or not at all) or we'd be taking debug info away from ourselves 🤔